sdk/storage/storage-file-share/test/fileserviceclient.spec.ts

```diff
@@ -380,4 +385,83 @@ describe("FileServiceClient", () => {
     assert.ok(typeof result.requestId);
     assert.ok(result.requestId!.length > 0);
   });
+
+  it("ListShares with deleted share", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
+    const shareClient = serviceClient.getShareClient(recorder.getUniqueName("share"));
+    await shareClient.create();
+    await shareClient.delete();
+
+    let found = false;
+    for await (const share of serviceClient.listShares({ includeDeleted: true })) {
+      if (share.name == shareClient.name) {
+        found = true;
+        assert.ok(share.version);
+        assert.ok(share.deleted);
+      }
+    }
+    assert.ok(found);
+  });
+
+  it("Undelete share positive", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
+    const shareClient = serviceClient.getShareClient(recorder.getUniqueName("share"));
+    await shareClient.create();
+    await shareClient.delete();
+
+    let found = false;
+    let shareDeleted: ShareItem | undefined;
+    for await (const share of serviceClient.listShares({ includeDeleted: true })) {
+      if (share.name == shareClient.name) {
+        found = true;
+        assert.ok(share.version);
+        assert.ok(share.deleted);
+        assert.ok(share.properties.deletedTime);
+        assert.ok(share.properties.remainingRetentionDays);
+
+        shareDeleted = share;
```

Comment:
I believe you will be able to remove a bunch of `!` from this file by doing the following here:

    shareDeleted = share!;

Reply:
shareDeleted here is initialized as undefined, so we can evaluate it's assigned when right share is found. Thanks for the suggestion anyway!

---
In reply to: [475540438](https://github.com/Azure/azure-sdk-for-js/pull/10792#discussion_r475540438) [](ancestors = 475540438)